q2_types/_feature_data.py

```diff
@@ -0,0 +1,183 @@
+                                                AlignedDNAFASTAFormat)
+
+
+def taxonomy_to_pandas_series(data_dir):
+    filepath = os.path.join(data_dir, 'taxonomy.tsv')
+    # Using read_csv for access to comment parameter, and set index_col,
+    # header, and parse_dates to Series.from_csv default settings for
+    # better handling of round-trips
+    df = pd.read_csv(filepath, sep='\t', comment='#', index_col=0, header=0,
+                     parse_dates=True, skip_blank_lines=True)
+    return df.iloc[:, 0]
+
+
+def taxonomy_to_qiime_metadata(data_dir):
+    filepath = os.path.join(data_dir, 'taxonomy.tsv')
+    df = pd.read_csv(filepath, sep='\t', comment='#', index_col=0, header=0,
+                     parse_dates=True, skip_blank_lines=True)
+    return qiime.Metadata(df)
+
+
+def pandas_series_to_taxonomy(view, data_dir):
+    filepath = os.path.join(data_dir, 'taxonomy.tsv')
+    view.to_csv(filepath, sep='\t', header=True, index=True)
+
+
+def qiime_metadata_to_taxonomy(view, data_dir):
+    filepath = os.path.join(data_dir, 'taxonomy.tsv')
+    df = view.to_dataframe()
+    df.to_csv(filepath, sep='\t', header=True, index=True)
+
+
+def sequences_to_pandas_series(data_dir):
+    index = []
+    seqs = []
+    for seq in skbio.io.read(os.path.join(data_dir, 'sequences.fasta'),
+                             format='fasta'):
+        index.append(seq.metadata['id'])
+        seqs.append(seq)
+    return pd.Series(data=seqs, index=index)
+
+
+def pandas_series_to_sequences(view, data_dir):
+    file = os.path.join(data_dir, 'sequences.fasta')
+
+    def generator():
+        yield from view
+
+    skbio.io.write(generator(), format='fasta', into=file)
+
+
+def aligned_dna_sequences_to_tabular_msa(data_dir):
+    return skbio.TabularMSA.read(
+        os.path.join(data_dir, 'aligned-dna-sequences.fasta'),
+        constructor=skbio.DNA, format='fasta')
+
+
+def tabular_msa_to_aligned_dna_sequences(view, data_dir):
+    return view.write(os.path.join(data_dir, 'aligned-dna-sequences.fasta'),
+                      format='fasta')
+
+
+plugin.register_data_layout(taxonomy_data_layout)
+plugin.register_data_layout_reader('taxonomy', 1, pd.Series,
+                                   taxonomy_to_pandas_series)
+plugin.register_data_layout_reader('taxonomy', 1, qiime.Metadata,
+                                   taxonomy_to_qiime_metadata)
+plugin.register_data_layout_writer('taxonomy', 1, pd.Series,
+                                   pandas_series_to_taxonomy)
+plugin.register_data_layout_writer('taxonomy', 1, qiime.Metadata,
+                                   qiime_metadata_to_taxonomy)
+
+plugin.register_data_layout(sequences_data_layout)
+plugin.register_data_layout_reader('sequences', 1, pd.Series,
+                                   sequences_to_pandas_series)
+plugin.register_data_layout_writer('sequences', 1, pd.Series,
+                                   pandas_series_to_sequences)
```

Comment:
I don't think so -- the `pd.Series` reader has the sequence ID (which is the feature ID) as its index, and the `skbio.Sequence` object as its value. What else would go in a `pd.DataFrame`?